app/lib/three_scale/spam_protection/checks/base.rb

```diff
@@ -1,31 +1,51 @@
+# frozen_string_literal: true
+
 module ThreeScale::SpamProtection
   module Checks
 
+    HIDE_STYLE = "position: absolute; width: 1px; height: 0; left: 0; overflow: hidden;"
+    SPAM_CHECKS_SECRET_KEY = -> { Rails.application.key_generator.generate_key('spam-protection-checks', 32) }
+
+    class SpamCheckError < StandardError; end
```

Comment:
I'm thinking maybe the error name should be more explicit?
Here one can think that some error happened when checking for spam, but we maybe we could call it `SpamDetectedError` or something like that to make it clear that it is thrown when we detect there was an spam attempt.

Reply:
It makes sense. Fixed